### ISSUE TITLE
Typescript: ClassInput does not include ViewStyle

### ISSUE THREAD

**Issue by jakobo** [jaredh159/tailwind-react-native-classnames#96]
The base input for `tw.style` does not seem to support `ViewStyle`. As a result, it's difficult to make components that pass a set of style props through as overrides. Forcibly recasting the style prop works, but it feels like the default styleprop should be built in. Using `ViewStyle` instead of the object definition will also provide autocomplete inside of `tw.style`. However, I'm not sure what the repercussions are of adding `ViewStyle` to the definitions of the class input. It works in testing, but I'm too new with Typescript to feel confident this is the right fix.

Reproduction:
```jsx
interface ComponentProps extends ViewProps {};

const Component: React.FC<ComponentProps> = ({style}) => {
  return <View style={tw.style("bg-red-400", style)} />
}
```

Error:
```
Argument of type 'StyleProp<ViewStyle>' is not assignable to parameter of type 'ClassInput'.
  Type 'ViewStyle' is not assignable to type 'ClassInput'.
    Type 'ViewStyle' is not assignable to type '{ [k: string]: string | number | boolean; }'.
      Index signature for type 'string' is missing in type 'ViewStyle'.ts(2345)
```

Workaround:
```jsx
interface ComponentProps extends ViewProps {};

const Component: React.FC<ComponentProps> = ({style}) => {
  return <View style={tw.style("bg-red-400", style as ClassInput)} />
}
```

**Comment by well1791** [comment]
@jakobo The problem is that type `ViewProps.styles` is different to `ClassInput` in the sense that `ClassInput` isn't even a subtype, so, instead of casting the type you must spread the `view.style` prop after the `tw.style` function.

```tsx
<View style={...tw.style("bg-red-400"), ...style} />
```

Now, if you instead want to pass `ClassInput` styles to your component, then you must change the component type so the `style` prop is set to `ClassInput`

```tsx
interface ComponentProps extends ViewProps {
  style: ClassInput;
};

const Component: React.FC<ComponentProps> = ({ style }) => {
  return <View style={tw.style("bg-red-400", style)} />
}
```

Haven't tested any of the samples above, but I believe they should work. Hope that helps.

**Comment by jakobo** [comment]
You can't extend and overwrite the `style` attribute to a non-compatible type unfortunately. You can do it with a union type, but it becomes unclear to the developer what type they should be using. This is critical for components like `Pressable` whose style prop also takes a function which is completely outside of the `ClassInput` typing.

Another reason you want to use `ViewStyle`, because that's what gives you hinting for any manual styles you add. It's typed in `Pressable` here https://github.com/facebook/react-native/blob/1b30dd074b579c2ae138a1111d07ddb56761315d/Libraries/Components/Pressable/Pressable.js#L140 as `ViewStyleProp | ((state: StateCallbackType) => ViewStyleProp)`.

To illustrate the problem, here's how I'm currently working around it by forcibly recasting. Again, the incoming `style` prop must be valid for `Pressable`, as you don't know how the end-developer will use it.

```jsx
interface TextButtonProps extends PressableProps {
  label?: string;
  // textStyle will behave identical to Pressable's style
  textStyle?: PressableProps["style"];
}

export const TextButton: React.FC<TextButtonProps> = ({
  label,
  style,
  textStyle,
  children,
  ...rest
}) => {
  return (
    <Pressable
      style={(state) =>
        tw.style(
          designSystem.getButtonStyles(state) as ClassInput,
          typeof style === "function"
            ? (style(state) as ClassInput)
            : (style as ClassInput)
        )
      }
      {...rest}
    >
      {(state) => (
        <>
          {label ? (
            <Text
              style={tw.style(
                designSystem.getTextButtonStyles(state) as ClassInput,
                typeof textStyle === "function"
                  ? (textStyle(state) as ClassInput)
                  : (textStyle as ClassInput)
              )}
            >
              {label}
            </Text>
          ) : (
            <>{children}</>
          )}
        </>
      )}
    </Pressable>
  );
};
```

For whatever reason, I can't get a custom `ClassInput` declaration to merge in a local `custom.d.ts`, otherwise I'd confirm the additional types fix the issue.

**Comment by jaredh159** [comment]
@jakobo yeah, the type of `ClassInput` is a little weak.  It is currently:

```ts
export type ClassInput =
  | string
  | string[]
  | boolean
  | null
  | undefined
  | { [k: string]: boolean | string | number };
```

That last line, with `{ [k: string]: boolean | string | number }` is a rough approximation of what a react native style object is, but you're right, it won't typecheck if you pass it something that TS knows is a `ViewStyle` -- although it will be happy with almost every valid object literal.  

I think that I can probably extend the type of `ClassInput`'s union to actually include `ViewStyle` to make this a little more ergonomic.  I'll do a little testing and hopefully ship a point release with this before too long.

Thanks for bringing this to my attention.

**Comment by jaredh159** [comment]
fixed in `v2.1.1` 👍
